### PR TITLE
Resurrect Bunny Dance

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This resolves https://lab.civicrm.org/dev/core/-/issues/3722 with a new approach to #23955 to prevent an infinite loop.

Before
-------------
In unit tests, `BunnyDance.mgd.php` was partially commented out, because uncommenting it gives:

```
Fatal error: Allowed memory size of 2147483648 bytes exhausted (tried to al  
  locate 20480 bytes) in /home/homer/buildkit/build/build-0/web/sites/all/mod  
  ules/civicrm/vendor/pear/pear-core-minimal/src/PEAR.php on line 898          
  Remove /home/homer/buildkit/build/build-0/web/sites/default/files/civicrm/e  
  xt/example-mixin                                                             
  Create /home/homer/buildkit/build/build-0/web/sites/default/files/civicrm/e  
  xt/example-mixin for scan-classes@1                                          
  Test scan-classes@1     
```

After
---------
Fatal infinite loop fixed.
Test fixed.

Technical Details
---------------
The problem #23955 wanted to solve was an infinite loop being caused by `Managed->reconcile()` calling itself.
This fixes that problem. And I've confirmed that existing tests do cover the functionality:
- `CRM_Case_BAO_CaseTypeForkTest.testManagerContact`
- `api_v3_CaseTypeTest.testCaseTypeDelete_InUse`

The `ManagedCaseTypeTest` turned out to be broken due to double-caching of the case types pseudoconstant, which was persisting beyond cache flushes. Fixed by removing the double-caching.

Also, while I was looking at it I optimized the `CRM_Case_ManagedEntities` reconcile and dropped some unused variables.

### CODE REVIEW

File: mixin/mgd-php@1/example/CRM/BunnyDance.mgd.php [hunk 1]
```diff
@@ -14,41 +14,40 @@
...
-        //   ],
-        // ],
+        'definition' => [
+          'activityTypes' => [
+            ['name' => 'Open Case', 'max_instances' => '1'],
+            ['name' => 'Follow up'],
+            ['name' => 'Change Case Type'],
+            ['name' => 'Change Case Status'],
+            ['name' => 'Change Case Start Date'],
+            ['name' => 'Link Cases'],
```
Comment (eileenmcnaughton) [inline mixin/mgd-php@1/example/CRM/BunnyDance.mgd.php:24]: This one should be 'Eat furniture'

Comment (colemanw) [reply mixin/mgd-php@1/example/CRM/BunnyDance.mgd.php:24]: Ah, I had a gnawing feeling that was the problem.

Comment (eileenmcnaughton) [reply mixin/mgd-php@1/example/CRM/BunnyDance.mgd.php:24]: lol

**Comment by colemanw** [comment]
retest this please

**Comment by demeritcowboy** [comment]
I'm not familiar with the bunny dance but the [hamster dance](https://www.youtube.com/watch?v=WEH2fk0ONag) was all the rage back when `<blink>` was cool.
Anyway nice job wrestling the bunny.
And I'll note xml files still work with this.

**Comment by totten** [comment]
Sweet. Not only does the bunny dance, but it also finds deep-gnarly bugs.

If the mixin-tests are passing, then I believe.

Nicely done @colemanw

**Pull request merged**